warehouse/core/src/main/java/datawave/util/CompositeTimestamp.java

```diff
@@ -0,0 +1,212 @@
+package datawave.util;
+
+import java.util.Calendar;
+import java.util.Comparator;
+import java.util.TimeZone;
+
+/**
+ * The composite timestamp allows us to encode two values in the timestamp to be used in accumulo keys. The event date will take the first (right hand most) 46
+ * bits. The last 17 bits (except for the sign bit) will be used to encode how many days after the event date that we should base the ageoff on. If the
+ * timestamp is negative, then to calculate the values the complement is taken and then the two values are extracted. The ageoff is encoded as an age off delta
+ * which is the number of days after the event date.
+ */
+public class CompositeTimestamp {
+
+    // The number of bits for the event date
+    private static final int allocationForEventDate = 46;
+    // A mask for the event date
+    private static final long mask = -1L >>> (8 * 8 - allocationForEventDate);
+    // The max age off delta in days
+    private static final long maxDiff = (-1L << (allocationForEventDate + 1)) >>> (allocationForEventDate + 1);
+    // A useful constant
+    public static final long MILLIS_PER_DAY = 1000 * 60 * 60 * 24;
+
+    // A constant for an invalid timestamp
+    public static final long INVALID_TIMESTAMP = Long.MIN_VALUE;
+    // The minimum event date
+    public static final long MIN_EVENT_DATE = (0 - mask); // also equivalent to Long.MIN_VALUE + 1
```

Comment:
Is this comment correct? If so, why not just initialize the min event date to Long.MIN_VALUE + 1?